ykbh/src/lib.rs

```diff
@@ -0,0 +1,267 @@
+        // FIXME Later this also sets other already initialised variables as well as the program
+        // counter of the interpreter.
+        let ptr = self.get_local(&Local(1)); // The trace inputs live in $1
+        unsafe {
+            std::ptr::copy(usize::to_ne_bytes(tio).as_ptr(), ptr as *mut u8, SUSIZE);
+        }
+    }
+
+    pub fn interpret(&mut self, body: &ykpack::Body) {
+        // Ignore yktrace::trace_debug.
+        if body.flags & ykpack::bodyflags::TRACE_DEBUG != 0 {
+            return;
+        }
+
+        for stmt in body.blocks[0].stmts.iter() {
+            match stmt {
+                Statement::MkRef(dest, src) => self.mkref(dest, src),
+                Statement::DynOffs { .. } => todo!("dynoffs"),
+                Statement::Store(dest, src) => self.store(dest, src),
+                Statement::BinaryOp { .. } => todo!("binop"),
+                Statement::Nop => {}
+                Statement::Unimplemented(_) | Statement::Debug(_) => todo!("Unimplemented"),
+                Statement::Cast(..) => todo!("cast"),
+                Statement::Call(..) | Statement::StorageDead(_) => unreachable!(),
+            }
+        }
+
+        match &body.blocks[0].term {
+            Terminator::Call {
+                operand: _op,
+                args: _args,
+                destination: _dest,
+            } => todo!(),
+            Terminator::Return => {}
+            t => todo!("{}", t),
+        }
+    }
+
+    /// Get the pointer to a Local.
+    fn get_local(&self, local: &Local) -> usize {
+        let offset = self.locals[usize::try_from(local.0).unwrap()];
+        self.map as usize + offset
+    }
+
+    /// Get the pointer for an IPlace, while applying all offsets.
+    fn iplace_to_ptr(&self, place: &IPlace) -> usize {
+        match place {
+            IPlace::Val {
+                local,
+                off,
+                ty: _ty,
+            } => {
+                // Get a pointer to the Val.
+                let dest_ptr = self.get_local(&local);
+                dest_ptr + usize::try_from(*off).unwrap()
+            }
+            IPlace::Indirect { ptr, off, ty: _ty } => {
+                // Get a pointer to the Indirect.
+                let dest_ptr = self.get_local(&ptr.local);
+                // Dereference the pointer by reading its value.
+                let deref_val =
+                    unsafe { &*std::ptr::slice_from_raw_parts(dest_ptr as *const u8, SUSIZE) };
+                // Turn the slice back into a usize.
+                let val = usize::from_ne_bytes(deref_val.try_into().unwrap());
+                // Add the offsets of the Indirect.
+                val + usize::try_from(ptr.off).unwrap() + usize::try_from(*off).unwrap()
+            }
+            _ => unreachable!(),
+        }
+    }
+
+    /// Write some bytes to an IPlace. The amount of bytes is determined by the type of the
+    /// destination.
+    fn write(&mut self, dest: &IPlace, src: *const u8) {
```

Comment:
```suggestion
    unsafe fn write(&mut self, dest: &IPlace, src: *const u8) {
```
Also prefer `&[u8]`. (It still has to be unsafe, as `dest` could contain a invalid or aliasing pointer.)